File: record-api/utils/RedisCache.py
import redis
from app_config import Config

class RedisCache:
    _client = None

    @classmethod
    def get_client(cls):
        if cls._client is None:
            cls._client = redis.Redis(
                host=Config.REDIS_HOST,
                port=Config.REDIS_PORT,
                decode_responses=True
            )
        return cls._client

    @classmethod
    def get(cls, key):
        return cls.get_client().get(key)

    @classmethod
    def set(cls, key, value, ttl=60):
        return cls.get_client().setex(key, ttl, value)

File: auth-api/src/Utils/RedisCache.php
<?php
namespace App\Utils;

use Predis\Client;
use App\Config\Config;

class RedisCache
{
    private static $client = null;

    public static function getClient(): Client
    {
        if (self::$client === null) {
            self::$client = new Client([
                'scheme' => 'tcp',
                'host' => Config::REDIS_HOST,
                'port' => Config::REDIS_PORT
            ]);
        }
        return self::$client;
    }

    public static function get(string $key)
    {
        return self::getClient()->get($key);
    }

    public static function set(string $key, $value, int $ttl = 300)
    {
        return self::getClient()->setex($key, $ttl, $value);
    }
}
